service/dap/server.go

```diff
@@ -1376,23 +1371,32 @@ func (s *Server) onEvaluateRequest(request *dap.EvaluateRequest) {
 			// As a shortcut also express the return values as a single string.
 			retVarsAsStr := ""
 			for _, v := range retVars {
-				retVarsAsStr += s.convertVariableToString(v, loadConfig) + ", "
+				retVarsAsStr += s.convertVariableToString(v) + ", "
 			}
 			response.Body = dap.EvaluateResponseBody{
 				Result:             strings.TrimRight(retVarsAsStr, ", "),
 				VariablesReference: s.variableHandles.create(&fullyQualifiedVariable{retVarsAsVar, "", false /*not a scope*/}),
 			}
 		}
 	} else { // {expression}
-		exprVar, err := s.debugger.EvalVariableInScope(goid, frame, 0, request.Arguments.Expression, loadConfig)
+		exprVar, err := s.debugger.EvalVariableInScope(goid, frame, 0, request.Arguments.Expression, DefaultLoadConfig)
 		if err != nil {
 			s.sendErrorResponseWithOpts(request.Request, UnableToEvaluateExpression, "Unable to evaluate expression", err.Error(), showErrorToUser)
 			return
 		}
+		// If we are in interactive mode (probably with sufficient space to display long strings),
+		// load the string with a more generous load config.
+		if ctxt := request.Arguments.Context; (ctxt == "repl" || ctxt == "variables") && exprVar.Kind == reflect.String {
+			if strVal := constant.StringVal(exprVar.Value); exprVar.Len > int64(len(strVal)) {
```

Comment:
I think just len(exprVal.Value) should work.

Reply:
exprVal.Value is constant.Value.